File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    
    vector<int> last;
    while(n > 0)
    {
        last.push_back(n % 3);
        n /= 3;
    }
    reverse(last.begin(), last.end());
    for(int i=0; i<last.size(); ++i)
        answer += last[i] * pow(3, i);
    
    return answer;
}

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.cs
using System;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        
        string str = "";
        while(n / 3 != 0)
        {
            str += n % 3;
            n /= 3;
        }
        str += n.ToString();
        char[] arr = str.ToCharArray();
        Array.Reverse(arr);
        for(int i = 0; i < arr.Length; ++i)
        {
            answer += (arr[i] - '0') * (int)Math.Pow(3, i);
        }
        
        
        return answer;
    }
}